pyfr/backends/openmp/kernels/batch-gemm.mako

```diff
@@ -16,7 +27,23 @@ void batch_gemm(const struct kargs *restrict args)
 {
     #pragma omp parallel for ${schedule}
     for (int ib = 0; ib < args->nblocks; ib++)
+    %if nfac == 1:
         args->exec(args->blockk,
                    args->b + ib*args->bblocksz,
                    args->c + ib*args->cblocksz);
+    %else:
+    {
+        fpdtype_t buf1[cblocksz], buf2[cblocksz];
+    %if nfac == 3:
```

Comment:
Make this a generic loop (the `loop.first`, `loop.last`, `loop.odd` and `loop.even` variables may be useful here).